graphbolt/src/csc_sampling_graph.cc

```diff
@@ -781,10 +783,16 @@ inline int64_t LaborPick(
     std::iota(picked_data_ptr, picked_data_ptr + num_neighbors, offset);
     return num_neighbors;
   }
-  torch::Tensor heap_tensor = torch::empty({fanout * 2}, torch::kInt32);
   // Assuming max_degree of a vertex is <= 4 billion.
-  auto heap_data = reinterpret_cast<std::pair<float, uint32_t>*>(
-      heap_tensor.data_ptr<int32_t>());
+  std::array<std::pair<float, uint32_t>, StackSize> heap;
+  auto heap_data = heap.data();
+  torch::Tensor heap_tensor;
+  if (fanout > StackSize) {
+    heap_tensor = torch::empty(
+        {fanout * (sizeof(heap_data[0]) / sizeof(int32_t))}, torch::kInt32);
```

Comment:
seems cause error when building on windows, `2` is ok for me as we explicitly declare `std::pair` above. Or try with `std::tuple_size<decltype(heap)::value_type>::value`?

Reply:
I just fixed it. It was related to casting of unsigned int to int.